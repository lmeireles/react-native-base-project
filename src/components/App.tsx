import React, { FC } from 'react'
import { NavigationContainer } from '@react-navigation/native'
import { I18nProvider } from 'providers/I18nProvider'
import ThemeProvider from 'providers/ThemeProvider'
import MainNavigation from 'views/MainNavigation'
import LoadingProvider from 'providers/LoadingProvider'
import AlertProvider from 'providers/AlertProvider'

const App: FC = () => (
  <ThemeProvider>
    <I18nProvider>
      <LoadingProvider>
        <AlertProvider>
          <NavigationContainer>
            <MainNavigation />
          </NavigationContainer>
        </AlertProvider>
      </LoadingProvider>
    </I18nProvider>
  </ThemeProvider>
)

export default App
