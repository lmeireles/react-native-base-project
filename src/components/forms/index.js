import InputHolder from './InputHolder'
import FormGroup from './FormGroup'
import Input from './Input'
import ErrorText from "./ErrorText";

export {
    InputHolder,
    FormGroup,
    ErrorText,
    Input,
};
