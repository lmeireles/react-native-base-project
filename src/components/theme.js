const colors = {
  white: "#EEE",
  light: "#B2B9C3",
  dark: "#26374B",
  primary: "#009A9A",
  secondary: "#FF2D66",
  danger: "#ff3f44"
};

const forms = {
  labelColor: colors.light,
  errorColor: colors.danger
};

const navigation = {
  headerBackground: colors.primary,
  tabBackground: colors.secondary,
  color: "#EEE",
  inactiveColor: "#EEEEEE88",
  iconSize: 28
};

const theme = Object.freeze({
  colors,
  forms,
  navigation,
  padding: 14,
  fontSize: 16
});

/*
  This function can be used to extract colors from components props
  Ex.: <MyComp primary />
  Inside "MyComp" we are able to call getColorFromProps(props) witch will return our theme`s primary color.
 */
export const getColorFromProps = (props) => {
  let finalColor = null;
  Object.keys(theme.colors).map(color => {
    if (color in props) {
      finalColor = theme.colors[color];
    }
  });

  return finalColor;
};

export default theme;
