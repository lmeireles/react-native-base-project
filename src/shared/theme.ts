import { Dimensions } from 'react-native'
import { DefaultTheme } from 'styled-components'
import * as styledComponents from 'styled-components/native'

export const {
  default: styled,
} = styledComponents as styledComponents.ReactNativeThemedStyledComponentsModule<DefaultTheme>

export const SMALL_DEVICE = Dimensions.get('window').width <= 350
export const MEDIUM_DEVICE = Dimensions.get('window').height <= 684
const UNIT = SMALL_DEVICE ? 4 : MEDIUM_DEVICE ? 6 : 8
const spacing = (factor: number): number => UNIT * factor

export const loadTheme = (externalTheme?: DefaultTheme): DefaultTheme => {
  return {
    spacing,
    viewPadding: SMALL_DEVICE ? spacing(2) : spacing(3),
    viewBackground: 'white',
    components: {
      textField: {
        height: 56,
        borderRadius: 12,
        background: 'gray07',
        labelColor: 'gray01',
        textColor: 'gray04',
      },
      ...(externalTheme?.components || {})
    },
    colors: {
      primary: '#240A36', // #364FD4
      secondary: '#e0e3fd',
      gray01: '#1E2126',
      gray02: '#5C6574',
      gray03: '#828B9A',
      gray04: '#B9C1CC',
      gray05: '#CAD1D9',
      gray06: '#E4E9ED',
      gray07: '#F0F3F8',
      white: '#FFFFFF',
      error: '#F13F11',
      success: '#00CC96',
      info: '#14C4E6',
      alert: '#FDC34E',
      ...(externalTheme?.colors || {})
    },
    text: {
      title: {
        family: 'Poppins-Bold',
        size: 28,
        spacing: 0.2,
      },
      smallTitle: {
        family: 'Poppins-Bold',
        size: 14,
      },
      subtitle: {
        family: 'Poppins-Regular',
        size: 16,
        spacing: 0.2,
      },
      body: {
        family: 'Poppins-Regular',
        size: 16,
        spacing: 0.1,
      },
      label: {
        family: 'Poppins-SemiBold',
        size: 16,
      },
      smallLabel: {
        family: 'Poppins-Regular',
        size: 12,
      },
      ...(externalTheme?.text || {})
    },
    buttons: {
      primary: {
        background: 'primary',
      },
      disabled: {
        background: 'gray05',
      },
      ...(externalTheme?.buttons || {})
    },
  }
}
