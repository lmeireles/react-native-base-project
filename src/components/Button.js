import React from 'react';
import {ActivityIndicator} from 'react-native';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import SimpleText from './SimpleText';
import theme, {getColorFromProps} from "./theme";

const ButtonContainer = styled.TouchableOpacity`
  width: 100%;
  padding: ${ p => p.padding}px;
  background: ${ p => p.backgroundColor};
  justify-content: center;
  align-items: center;
  border-radius: ${p => p.round ? 7 : 2}px;
`;

const Button = (props) => {
  const {loading, title, color, ...rest} = props;

  rest.backgroundColor = getColorFromProps(props) || rest.backgroundColor;

  return (
    <ButtonContainer {...rest}>
      { loading ?
        <ActivityIndicator color={color} />
        :
        <SimpleText color={color} bold>{title}</SimpleText>
      }
    </ButtonContainer>
  )
};

Button.propTypes = {
  backgroundColor: PropTypes.string,
  color: PropTypes.string,
  padding: PropTypes.number,
  round: PropTypes.bool,
  loading: PropTypes.bool,
  title: PropTypes.string
};

Button.defaultProps = {
  backgroundColor: theme.colors.light,
  color: '#EEE',
  padding: theme.padding,
  round: false,
  loading: false,
};

export default Button;
