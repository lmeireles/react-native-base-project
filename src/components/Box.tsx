import React, { FC } from 'react'
import {
  View,
  Animated,
  StyleSheet,
  ViewProps,
  FlexStyle,
  TouchableOpacity,
} from 'react-native'
import { DefaultTheme, ThemeColors, useTheme } from 'styled-components'
import { computeLayoutStyles, LayoutProps } from "shared/layout";

interface BoxInnerProps extends LayoutProps {
  radius?: number
  position?: 'relative' | 'absolutefill' | 'absolute'
  height?: number
  width?: number | string

  // false = flex: undefined, true = flex: 1
  flex?: FlexStyle['flex']
  flexFlow?: FlexStyle['flexDirection']
  justifyContent?: FlexStyle['justifyContent']
  alignItems?: FlexStyle['alignItems']
  color?: keyof ThemeColors
  borderColor?: keyof ThemeColors
  borderWidth?: any
  shadow?:boolean
}

const computeBoxStyles = (
  inp: BoxInnerProps,
  theme: DefaultTheme,
  style: any
) => [
  computeLayoutStyles(inp, theme),
  {
    flexDirection: inp.flexFlow,
    position: inp.position !== 'absolutefill' ? inp.position : undefined,
    alignItems: inp.alignItems,
    alignSelf: inp.alignSelf,
    justifyContent: inp.justifyContent,
    height: inp.height,
    width: inp.width,
    borderWidth: inp.borderWidth,
    borderColor:
      inp.borderColor != null ? theme.colors[inp.borderColor] : undefined,
    borderRadius:
      inp.radius !== undefined ? theme.spacing(inp.radius) : undefined,
    backgroundColor: inp.color != null ? theme.colors[inp.color] : undefined,
  },
  inp.position === 'absolutefill' && StyleSheet.absoluteFill,
  inp.shadow && {
    shadowColor: theme.colors.gray01,
    shadowRadius: 10,
    shadowOpacity: .1,
    elevation: 1,
  },
  style,
]

export type BoxProps = ViewProps & BoxInnerProps

export const Box: FC<BoxProps> = props => {
  const theme = useTheme()
  const style = computeBoxStyles(props, theme, props.style)

  return <View {...props} style={style} />
}

export const ABox: FC<
  Omit<ViewProps, 'style'> & { style: any } & BoxInnerProps
> = props => {
  const theme = useTheme()
  const style = computeBoxStyles(props, theme, props.style)

  return <Animated.View {...props} style={style} />
}

export const TouchBox: FC<
  BoxProps & { onPress: () => void; disabled?: boolean }
> = props => {
  const theme = useTheme()
  const style = computeBoxStyles(props, theme, props.style)

  return <TouchableOpacity {...props} style={style} />
}
