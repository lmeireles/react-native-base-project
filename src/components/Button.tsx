import React, { FC } from 'react'
import { ButtonVariants, useTheme } from 'styled-components'
import {
  StyleProp,
  TouchableOpacity,
  TouchableOpacityProps,
  ViewStyle,
} from 'react-native'
import { Text } from 'components/Text'
import { computeLayoutStyles, LayoutProps } from 'shared/layout'

interface ButtonProps extends TouchableOpacityProps {
  variant?: keyof ButtonVariants
}
const Button: FC<ButtonProps & LayoutProps> = ({
  variant = 'primary',
  disabled,
  children,
  style,
  ...props
}) => {
  const theme = useTheme()
  const btnStyle = theme.buttons[disabled ? 'disabled' : variant]
  const hasBorder = !!btnStyle.borderWidth && !!btnStyle.borderColor

  const styles = [
    computeLayoutStyles(props, theme),
    {
      alignItems: 'center',
      justifyContent: 'center',
      width: '100%',
      height: 54,
      borderRadius: 12,
      backgroundColor: theme.colors[hasBorder ? 'white' : btnStyle.background],
      borderWidth: hasBorder ? btnStyle.borderWidth : 0,
      borderColor: theme.colors[btnStyle.borderColor || 'primary'],
    } as StyleProp<ViewStyle>,
    style,
  ]

  return (
    <TouchableOpacity {...props} disabled={disabled} style={styles}>
      <Text
        variant="label"
        color={hasBorder ? btnStyle.borderColor || 'primary' : 'white'}
      >
        {children}
      </Text>
    </TouchableOpacity>
  )
}

export default Button
