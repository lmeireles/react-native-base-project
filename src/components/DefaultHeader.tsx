import React, { FC, memo } from 'react'
import { Box, TouchBox } from './Box'
import { useSafeAreaInsets } from 'react-native-safe-area-context'
import { Text } from './Text'
import { useNavigation } from '@react-navigation/native'
import { useTheme } from 'styled-components'

interface DefaultHeader {
  title: string
  hideLeft?: boolean
  customLeft?: () => void
}
const DefaultHeader: FC<DefaultHeader> = ({ title, hideLeft, customLeft }) => {
  const theme = useTheme()
  const insets = useSafeAreaInsets()
  const navigation = useNavigation()

  const handleLeftAction = () => {
    if (customLeft) {
      customLeft()
    } else {
      navigation.goBack()
    }
  }

  return (
    <Box
      color="white"
      width="100%"
      style={{
        paddingTop: insets.top,
        shadowColor: theme.colors.primary,
        shadowRadius: 20,
        shadowOpacity: 0.15,
      }}
    >
      <Box pv={2} flexFlow="row" alignItems="center">
        {hideLeft ? (
          <Box pr={2.2} width={30} />
        ) : (
          <TouchBox pl={2.2} width={30} onPress={handleLeftAction}>
            Back
          </TouchBox>
        )}

        <Box flex={1}>
          <Text variant="title" align="center">
            {title}
          </Text>
        </Box>

        <Box pr={2.2} width={30} />
      </Box>
    </Box>
  )
}

export default memo(DefaultHeader)
