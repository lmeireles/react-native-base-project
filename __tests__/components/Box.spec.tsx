import React from 'react'
import { Box } from 'components/Box'
import { render, snapshot } from '../test-utils'
import { loadTheme } from 'shared/theme'
import 'jest-styled-components/native'
import { ThemeColors } from 'styled-components'

describe('Box snapshots', () => {
  const theme = loadTheme()

  it('default', () => {
    snapshot(render(<Box />))
  })

  it('colors', () => {
    Object.keys(theme.colors).map(key => {
      snapshot(render(<Box color={key as keyof ThemeColors} />))
    })
  })

  it('dimensions', () => {
    snapshot(render(<Box m={3} pt={5} ph={6} height={5} width={100} />))
  })

  it('flex', () => {
    snapshot(
      render(<Box flex={5} flexFlow="column" justifyContent="flex-end" />)
    )
  })
})

describe('Box matches theme', () => {
  test('render colors', () => {
    const theme = loadTheme()

    Object.keys(theme.colors).map(key => {
      const { toJSON } = render(<Box color={key as keyof ThemeColors} />)
      // @ts-ignore
      expect(toJSON()).toHaveStyleRule(
        'backgroundColor',
        theme.colors[key as keyof ThemeColors]
      )
    })
  })
})
