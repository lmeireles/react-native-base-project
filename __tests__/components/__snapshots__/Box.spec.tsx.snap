// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box snapshots colors 1`] = `
<View
  color="primary"
  style={
    Array [
      Object {
        "alignSelf": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": "#240A36",
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "position": undefined,
        "width": undefined,
      },
      false,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`Box snapshots colors 2`] = `
<View
  color="secondary"
  style={
    Array [
      Object {
        "alignSelf": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": "#e0e3fd",
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "position": undefined,
        "width": undefined,
      },
      false,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`Box snapshots colors 3`] = `
<View
  color="gray01"
  style={
    Array [
      Object {
        "alignSelf": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": "#1E2126",
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "position": undefined,
        "width": undefined,
      },
      false,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`Box snapshots colors 4`] = `
<View
  color="gray02"
  style={
    Array [
      Object {
        "alignSelf": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": "#5C6574",
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "position": undefined,
        "width": undefined,
      },
      false,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`Box snapshots colors 5`] = `
<View
  color="gray03"
  style={
    Array [
      Object {
        "alignSelf": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": "#828B9A",
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "position": undefined,
        "width": undefined,
      },
      false,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`Box snapshots colors 6`] = `
<View
  color="gray04"
  style={
    Array [
      Object {
        "alignSelf": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": "#B9C1CC",
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "position": undefined,
        "width": undefined,
      },
      false,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`Box snapshots colors 7`] = `
<View
  color="gray05"
  style={
    Array [
      Object {
        "alignSelf": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": "#CAD1D9",
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "position": undefined,
        "width": undefined,
      },
      false,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`Box snapshots colors 8`] = `
<View
  color="gray06"
  style={
    Array [
      Object {
        "alignSelf": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": "#E4E9ED",
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "position": undefined,
        "width": undefined,
      },
      false,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`Box snapshots colors 9`] = `
<View
  color="gray07"
  style={
    Array [
      Object {
        "alignSelf": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": "#F0F3F8",
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "position": undefined,
        "width": undefined,
      },
      false,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`Box snapshots colors 10`] = `
<View
  color="white"
  style={
    Array [
      Object {
        "alignSelf": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": "#FFFFFF",
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "position": undefined,
        "width": undefined,
      },
      false,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`Box snapshots colors 11`] = `
<View
  color="error"
  style={
    Array [
      Object {
        "alignSelf": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": "#F13F11",
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "position": undefined,
        "width": undefined,
      },
      false,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`Box snapshots colors 12`] = `
<View
  color="success"
  style={
    Array [
      Object {
        "alignSelf": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": "#00CC96",
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "position": undefined,
        "width": undefined,
      },
      false,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`Box snapshots colors 13`] = `
<View
  color="info"
  style={
    Array [
      Object {
        "alignSelf": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": "#14C4E6",
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "position": undefined,
        "width": undefined,
      },
      false,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`Box snapshots colors 14`] = `
<View
  color="alert"
  style={
    Array [
      Object {
        "alignSelf": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": "#FDC34E",
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "position": undefined,
        "width": undefined,
      },
      false,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`Box snapshots default 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": undefined,
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "position": undefined,
        "width": undefined,
      },
      false,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`Box snapshots dimensions 1`] = `
<View
  height={5}
  m={3}
  ph={6}
  pt={5}
  style={
    Array [
      Object {
        "alignSelf": undefined,
        "margin": 24,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": 48,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": 40,
        "paddingVertical": undefined,
      },
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": undefined,
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": undefined,
        "flexDirection": undefined,
        "height": 5,
        "justifyContent": undefined,
        "position": undefined,
        "width": 100,
      },
      false,
      undefined,
      undefined,
    ]
  }
  width={100}
/>
`;

exports[`Box snapshots flex 1`] = `
<View
  flex={5}
  flexFlow="column"
  justifyContent="flex-end"
  style={
    Array [
      Object {
        "alignSelf": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
      Object {
        "alignItems": undefined,
        "alignSelf": undefined,
        "backgroundColor": undefined,
        "borderColor": undefined,
        "borderRadius": undefined,
        "borderWidth": undefined,
        "flexDirection": "column",
        "height": undefined,
        "justifyContent": "flex-end",
        "position": undefined,
        "width": undefined,
      },
      false,
      undefined,
      undefined,
    ]
  }
/>
`;
